### PR TITLE
build: provide automatic changelog generation (#7)

### PR DESCRIPTION
**Pull request by csshugs**
updated dependencies, run `npm install`.

### CODE REVIEW

**Comment by csshugs** [comment]
You can test the automatic changelog generation by running `npm run release`. **Be aware though** that you reset the automatically created commit and delete the created Git tag afterwards.

File: package.json [hunk 1]
```diff
@@ -11,7 +11,53 @@
     "start": "npm-run-all -p -c server sass:watch",
     "server": "ws --directory test/",
     "sass:compile": "node-sass test/style.scss test/style.css",
-    "sass:watch": "npm-run-all sass:compile \"sass:compile -- -w\""
+    "sass:watch": "npm-run-all sass:compile \"sass:compile -- -w\"",
+    "release": "standard-version -t ''"
+  },
+  "standard-version": {
```
Comment (fuhlig) [inline package.json]: Add the config into a dedicated style (separation of concerns) to not bloat the `package.json`

File: package.json [hunk 2]
```diff
@@ -11,7 +11,53 @@
...
+        "type": "build",
+        "hidden": true
+      },
+      {
+        "type": "ci",
+        "hidden": true
+      },
+      {
+        "type": "docs",
+        "section": "Features"
```
Comment (fuhlig) [inline package.json]: wrong section for `docs`?

File: package.json [hunk 3]
```diff
@@ -38,6 +84,7 @@
     "node-sass": "^4.12.0",
     "npm-run-all": "^4.1.5",
     "rimraf": "^3.0.0",
-    "sassdoc": "^2.7.1"
+    "sassdoc": "^2.7.1",
+    "standard-version": "^7.0.0"
```
Comment (fuhlig) [inline package.json]: Avoid using `^` and `~` in `package.json` and use fixed versions instead.
Despite `^` being the default option, set it to fixed versions with `.npmrc` and `save-exact=true`

```
// .npmrc
save-exact=true
```

Not part of this PR, but consider creating a separate PR to do it. In order to update, use a tool like [ncu](https://www.npmjs.com/package/npm-check-updates)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -19,3 +19,19 @@ Allowed types:
 - **test**: Adding missing tests or correcting existing tests
 
 _(based on [Angular contribution guidelines](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type))_
+
+## Release a new version
```
Comment (fuhlig) [inline CONTRIBUTING.md:105]: This workflow looks far from automated.
Despite not having a monorepo, checkout the [lerna publish workflow](https://github.com/lerna/lerna/tree/master/commands/publish#readme) to get an idea of a automated release

**Review by fuhlig — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (csshugs) [reply package.json]: Good question. I asked myself, if an extra “Documentation” entry in the changelog is really necessary. Because added docs is always a feature, too. But maybe you’re right. It doesn’t hurt to add all the docs-related changes to a separate section in the changelog. 👍

File: CONTRIBUTING.md (hunk 4 above)
Comment (csshugs) [reply CONTRIBUTING.md:105]: I agree that this is not 100% automated, but IMO, we should not introduce a 100% automation for this at the moment, without testing it with this approach for a while. Specifically an automated git push to remote can really complicate things, when stuff has to be reverted. I would always check the automated changelog file before pushing, just to see if the formatting is correct. Otherwise this could lead to a mis-formatted markdown file, which is not ideal for the users.

Please bear in mind, that we don’t add the changelog entries manually alongside each pull request, where the formatting would be under control of the review process. When the creation of the changelog itself is done **during** the release process, we have no chance of controlling the formatting of that and there is a chance that we need to redo the changes, reverting thins in git and do the whole thing manually anyway.

If you look to the list of tasks to do for a new release, steps 1–3 and 8–10 have to be done manually anyway.

File: .nvmrc [hunk 5]
```diff
@@ -1 +1,2 @@
 10
+save-exact=true
```
Comment (csshugs) [inline .nvmrc]: Wrong file. This needs to go into `.npmrc` not `.nvmrc`.

**Review by fuhlig — APPROVED** [review]
